### PR TITLE
Add picard dedup

### PR DESCRIPTION
**Pull request by cagaser**
### Checklist
- [x] Pull request details were added to CHANGELOG.md.
- [ ] Documentation was updated (if required).
- [x] `parameter_meta` was added/updated (if required).

### CODE REVIEW

File: umi.wdl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        import pysam 
+        import sys
+        import os
+
+        from typing import Tuple
+
+        def split_umi_from_name(name) -> Tuple[str, str]:
+            id_and_rest = name.split(maxsplit=1)
+            if len(id_and_rest) == 1:
+                id, = id_and_rest
```
Comment (DavyCats) [inline umi.wdl:49]: Is this line correct?

Comment (rhpvorderman) [reply umi.wdl:49]: Yes. `id_and_rest` is a result of `name.split(maxsplit=1)`. This has two possible outcomes.
1. The name does not contain whitespace `My_illumina_read:klkd:42342:sads_AGGCTACG`
2. The name does contain whitespace: `asdasd:kalkjsal:asdjka_AGCGTAGC nanoporeftw anothercomment`

The `len` check tests for case 1. In which case `id_and_rest` will be `["My_illumina_read:klkd:42342:sads_AGGCTACG"]`. A list of one item. By doing `id, = id_and_rest`, the list is unpacked here into the single item ID. This is basic tuple unpacking for one item. 

If you feel it is more readable `id = id_and_rest[0]` accomplishes the same thing. I did it in the aforementioned way to keep symmetry with the other case where the list has two items.

Alternatively the code could be
```python
id_and_rest = name.split(maxsplit=1)
id = id_and_rest[0]  # name.split wll always give at least one item.
rest = id_and_rest[1] if len(id_and_rest) > 1 else ""
```
That is perhaps more clear.

Comment (rhpvorderman) [reply umi.wdl:49]: I keep the code here: https://git.lumc.nl/rhpvorderman/bam-umi-annotate. Including a few tests for the name part.

Comment (rhpvorderman) [reply umi.wdl]: I updated the code to be more simple and conventional.

**Review by DavyCats — APPROVED** [review]

**Comment by rhpvorderman** [comment]
@DavyCats could you have a look at the changes since? It seems to run in production well now, using WGS files.

File: umi.wdl [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        File inputBam
+        String outputPath = "output.bam"
+        String umiTag = "RX"
+
+        String memory = "2G"
+        Int timeMinutes = 1 + ceil(size([inputBam], "G") * 10)
+        String dockerImage = "quay.io/biocontainers/pysam:0.17.0--py39h051187c_0"
+    }
+    String bamIndexPath = sub(select_first([outputPath]), "\.bam$", ".bai")
+    command <<<
```
Comment (DavyCats) [inline umi.wdl:38]:
```suggestion
    }
    
    String bamIndexPath = sub(select_first([outputPath]), "\.bam$", ".bai")
    
    command <<<
```

**Review by DavyCats — APPROVED** [review]

File: umi.wdl (hunk 2 above)
Comment (rhpvorderman) [reply umi.wdl]: Time to integrate those linters!

**Pull request merged**